cmd/prune-junit-xml/prunexml.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/xml"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+)
+
+// JUnitTestSuites is a collection of JUnit test suites.
```

Comment:
// TODO: should we re-use definitions for these from somewhere? What happens when the other tools produce files with more fields we don't have here?

Reply:
I suppose we could decode into a `interface{}` and then drill down by following the field names. The code is going to be more complicated, but I agree that loosing unknown fields doesn't look good.

Reply:
It would be similar to the `findStrings` function in https://github.com/kubernetes/kubernetes/pull/108458/files

Reply:
@BenTheElder i swiped them from gotestsum, which has not touched them for a while.

@BenTheElder @pohly yes, we can make it better as needed, i'd like to land this first and iterate please.

Reply:
Some of the packages support this other trick with an inline map in the struct with other named fields:
https://github.com/kubernetes-sigs/kind/blob/412a2adaeb3d86eadba93010cf72d97c1a0857c6/pkg/cluster/internal/kubeconfig/internal/kubeconfig/types.go#L86-L88
`OtherFields map[string]interface{} `yaml:",inline,omitempty"`

It might just be the yaml lib though ... trying to recall

Reply:
@BenTheElder there are ways to make this better, but this was the most straight forward of the options i looked at.

yes for sure yaml has that, see the `yamlfmt` utility

Reply:
> @BenTheElder i swiped them from gotestsum, which has not touched them for a while.

can we please leave some note about this? gotestsum has not touched them for a while but I have a nagging feeling that someday it will and someone will be confused about missing fields 🙃